src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,296 @@
+import { styles } from "./styles";
+import { TransactionsByCategory, Transaction } from "./TransactionsByCategory";
+import { ShowFullListToggle } from "./ShowFullListToggle";
+import {
+  DistantTransactionTitle,
+  RecentTransactionTitle,
+  NoPreviousTransactionTitle,
+  UsageQuotaTitle
+} from "./TransactionTitle";
+import { AppealButton } from "./AppealButton";
+import { CampaignConfigContext } from "../../../context/campaignConfig";
+import { ProductContext } from "../../../context/products";
+import { AuthContext } from "../../../context/auth";
+import { Quota, PastTransactionsResult, CampaignPolicy } from "../../../types";
+import { AlertModalContext, systemAlertProps } from "../../../context/alert";
+
+const DURATION_THRESHOLD_SECONDS = 60 * 10; // 10 minutes
+const MAX_TRANSACTIONS_TO_DISPLAY = 5;
+const BIG_NUMBER = 99999;
+
+interface NoQuotaCard {
+  ids: string[];
+  cart: Cart;
+  onCancel: () => void;
+  onAppeal?: () => void;
+  quotaResponse: Quota | null;
+}
+
+export interface TransactionsByCategoryMap {
+  [category: string]: {
+    transactions: Transaction[];
+    order: number;
+    hasLatestTransaction: boolean;
+  };
+}
+
+const sortTransactionsByOrder = (
+  a: TransactionsByCategory,
+  b: TransactionsByCategory
+): number => a.order - b.order;
+
+/**
+ * Given past transactions, group them by categories.
+ *
+ * @param sortedTransactions Past transaction results sorted by transaction time in desc order
+ * @param allProducts Policies
+ * @param latestTransactionTime Transaction time of latest transaction
+ */
+export const groupTransactionsByCategory = (
+  sortedTransactions: PastTransactionsResult["pastTransactions"] | null,
+  allProducts: CampaignPolicy[],
+  latestTransactionTime: Date | undefined
+): TransactionsByCategoryMap => {
+  const transactionsByCategoryMap: TransactionsByCategoryMap = {};
+  sortedTransactions?.forEach(item => {
+    const policy = allProducts?.find(
+      policy => policy.category === item.category
+    );
+    const categoryName = policy?.name ?? item.category;
+    const formattedDate = format(item.transactionTime, "d MMM yyyy, h:mma");
+
+    if (!(categoryName in transactionsByCategoryMap)) {
+      transactionsByCategoryMap[categoryName] = {
+        transactions: [],
+        hasLatestTransaction: false,
+        order: policy?.order ?? BIG_NUMBER
+      };
+    }
+    transactionsByCategoryMap[categoryName].transactions.push({
+      transactionDate: formattedDate,
+      details: getAllIdentifierInputDisplay(item.identifierInputs ?? []),
+      quantity: formatQuantityText(
+        item.quantity,
+        policy?.quantity.unit || { type: "POSTFIX", label: " qty" }
+      ),
+      isAppeal: policy?.categoryType === "APPEAL"
+    });
+    transactionsByCategoryMap[categoryName].hasLatestTransaction =
+      transactionsByCategoryMap[categoryName].hasLatestTransaction ||
+      differenceInSeconds(latestTransactionTime || 0, item.transactionTime) ===
+        0;
```

Comment:
Would checking if `latestTransactionTime === item.transactionTime` suffice?

Reply:
No, because in the backend we add a millisecond to each item in the same transaction. So need to round to seconds to check instead~